content/PIV/Guides/Securing_SSH_with_OpenPGP_or_PIV.adoc

```diff
@@ -0,0 +1,38 @@
+== Securing SSH with OpenPGP or PIV
+SSH supports multiple types of asymmetric certificates when implementing secure communication with remote systems, including OpenPGP format and X.509 (PIV). The YubiKey supports both certificate formats, but typically deployment considerations or the particular use case will dictate the format.

```

Comment:
Thinking of OpenPGP and PIV as certificate formats is a bit confusing. Not sure that OpenPGP even have certificates, actually. 

Can we write something along the lines of the following?

Secure communication with remote systems with the help of a YubiKey can be done either with the YubiKey’s PIV application using, for example, OpenSSH, or with the YubiKey’s PGP application. Typically, deployment considerations or the particular use case will dictate what application to use

Reply:
@FFincannon please check my grammar, but how does this sound (heavily based off of Aveen's feedback):

Secure communication with remote systems via SSH can be protected using key-based authentication with the user's key residing on a physical YubiKey device. The YubiKey offers two options when securing a user's SSH key data: OpenPGP and PIV. Typically, deployment considerations or the particular use case will dictate what application to use.

beyond username with the help of a YubiKey can be done either with the YubiKey’s PIV application using, for example, OpenSSH, or with the YubiKey’s PGP application. Typically, deployment considerations or the particular use case will dictate what application to use

Reply:
A key point to help someone choose:
If you use the OpenPGP option for SSH then SSH will use the same key as GIT would use to sign a commit (since GIT uses  the gpg command to sign). However, the OpenPGP option is more work to set up properly (ssh will speak to the gpg agent) so the PIV option would probably be easier for anyone who doesn't use GIT or who actually wants to use a different key for SSH and GIT. 

Furthermore (maybe repeating stuff you already know) for the PIV option one my use the opensc pkcs11 module, or our new ykcs11 module. Our ykcs11 is a lighter software package, so would probably be an easier path for someone who doesn't already have the opensc software installed. (The same goes for OpenPGP vs PIV as well, if you already have gpg working)

Reply:
So for me as a developer the OpenPGP path is the natural one, I need gpg working to be able to use git, and I only have to upload a single public key to github, which it uses to both authenticate my ssh access (used for git pushing) and for verifying the signatues on my git commits.

Reply:
Another detail: ykcs11 gives access to all 24 slots on the PIV application, while opensc only exposes 4. Additionally, the attestation certificates are only available with ykcs11.

Reply:
Even as a non-developer, I like to have a few concrete reasons for choosing one way or another. The new introductory paragraph reads:

Securing communication with remote systems via SSH can be done by using key-based authentication with the user's key residing on a physical YubiKey. With a YubiKey, the user's SSH key data can be secured using OpenPGP and/or PIV. Typically, deployment considerations or the particular use case will dictate which application to use.

If you use the OpenPGP option for SSH, then SSH will use the same key as Git would use to sign a commit (since Git uses the GnuPG command to sign). However, the OpenPGP option is more complicated to set up properly (SSH will speak to the GnuPG agent) so the PIV option would probably be easier for anyone who does not use Git or who does not want to use the same key for both SSH and Git. Furthermore, for the PIV option, you can use the opensc-pkcs11 module, or our new ykcs11 module. Our ykcs11 is a lighter software package, so it would probably offer an easier path for someone who does not already have the OpenSC software installed. (The same goes for OpenPGP vs PIV as well, if you already have GnuPG working).

For a developer the OpenPGP path is typically the natural one, because GnuPG needs to be working to enable the use of Git, and it is only necessary to upload a single public key to GitHub, which it then uses both to authenticate SSH access (used for Git pushing) and for verifying the signatures on Git commits.